Update authorization client error search and handle 500 errors

JuanCrg90 opened the pull request:
This updates the authorization client to use the same approach that api client uses to search for the error class. This adds a new ErrorParser class that is in charge to verify which error class should be returned. Also this adds a status code validation to raise a EasyMeli exception in server side errors.

JuanCrg90 committed:
Update Authorization client to use error list

This updates the authorization client to use the same approach that api
client uses to search for the error class.

JuanCrg90 committed:
Refactor Error class search to its own class

To avoid duplication the error class method has been moved to Error
Parse class

JuanCrg90 committed:
Bump version

enortham commented on lib/easy_meli/errors.rb:
```diff
@@ -73,4 +73,23 @@ def local_message
+      'unknown_error' => EasyMeli::UnknownError
+    }
+
+  def self.error_class(body)
```
Bad indents but I would also move this into its own file. I think this file should be only for errors.

enortham reviewed (changes requested):
I notice that you're sending lots of versions. FYI you can test the gem locally with real data to make sure it works as expected.

JuanCrg90 committed:
Add server side error validation

Some server side error are returned without body. This adds a EasyMeli
Error class for the common 500 errors.

JuanCrg90 commented on lib/easy_meli/error_parser.rb:
```diff
@@ -0,0 +1,38 @@
+    'unknown_error' => EasyMeli::UnknownError
+  }
+
+  STATUS_ERRORS = {
```
I found that some ML server side errors return an empty body. I've created specific errors that inherit from `EasyMeli:: ServerError` to help with the error handling.

JuanCrg90 commented:
@enortham changes addressed and I've added some new error classes for server side errors, I found that some  500 responses are sent without body, I think is better if EasyMeli handles these responses and raises an exception.

enortham commented on lib/easy_meli/error_parser.rb:
```diff
@@ -0,0 +1,38 @@
+  def self.status_error_class(response)
+    return unless self.server_side_error?(response)
+
+    STATUS_ERRORS[response.code]
```
Maybe you should some general sort of error if the code doesn't match anything.

JuanCrg90 commented:
Updated @enortham what do you think?
